crates/primitives/src/transaction/tx_type.rs

```diff
@@ -45,6 +45,9 @@ pub enum TxType {
 }
 
 impl TxType {
+    /// The max type reserved by an EIP.
+    pub const MAX_RESERVED_EIP: TxType = Self::EIP4844;
```

Comment:
this should include `ETHEREUM` because other networks have more types (deposit e.g.)

Reply:
do other networks also have EIPs? I put that in the name for this reason